src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,214 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(T value) {
+            this.value = value;
+            this.next = null;
+            this.prev = null;
+        }
+    }
+
+    void checkIndexValid(int index) {
+        if (index < 0) {
+            throw new IndexOutOfBoundsException("Index must be greater then 0 but was :" + index);
+        }
+
+        if (index > size - 1) {
+            throw new IndexOutOfBoundsException("Index can't be higher then size but Index was :"
+                    + index + " and Size was " + size + " no such index in Node");
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
```

Comment:
Don't complicate if-else construction. Consider creating a `findNodeByIndex` method to avoid code duplication and simplify the `add(T value, int index)` method.